PhysicsTools/PatAlgos/plugins/PATMuonProducer.cc

```diff
@@ -815,6 +836,94 @@ double PATMuonProducer::getRelMiniIsoPUCorrected(const pat::Muon& muon, float rh
   return pat::muonRelMiniIsoPUCorrected(muon.miniPFIsolation(), muon.p4(), drcut, rho);
 }
 
+
+double PATMuonProducer::puppiCombinedIsolation(const pat::Muon& muon, const pat::PackedCandidateCollection *pc)
+{
+  double dR_threshold = 0.4;
+  double dR2_threshold = dR_threshold * dR_threshold ;
+  double mix_fraction = 0.5;
+  const std::vector<pat::PackedCandidate> * candidates = pc;
+  enum particleType{
+      CH = 0,
+      NH = 1,
+      PH = 2,
+      OTHER = 100000
+    };
+  double val_PuppiWithLep    [3]= {0,0,0} ;
+  double val_PuppiWithoutLep [3]= {0,0,0} ;  
+
+  for( std::vector<pat::PackedCandidate>::const_iterator cand = candidates -> begin();cand != candidates ->end();cand ++ ){//pat::pat::PackedCandidate loop start
+     const particleType pType =
+        isChargedHadron( cand -> pdgId() ) ? CH :
+        isNeutralHadron( cand -> pdgId() ) ? NH :
+        isPhoton( cand -> pdgId() ) ? PH : OTHER ;
+     if( pType == OTHER ){
+        if( cand -> pdgId() != 1 && cand -> pdgId() != 2
+        && abs( cand -> pdgId() ) != 11
+        && abs( cand -> pdgId() ) != 13){
+        LogTrace("PATMuonProducer") <<"candidate with PDGID = " << cand -> pdgId() << " is not CH/NH/PH/e/mu or 1/2 (and this is removed from isolation calculation)" << std::endl;
+       }
+       continue;
+     }
+     
+     double d_eta = fabs( cand->eta() - muon.eta() ) ;
+
+     if( d_eta > dR_threshold ) continue ;
+
+     double d_phi = fabs(reco::deltaPhi(cand->phi(),muon.phi()));
+
+     if( d_phi > dR_threshold ) continue ;
+
+
+     double dR2 = d_eta * d_eta  + d_phi * d_phi ;
+     if(  dR2 > dR2_threshold ) continue ;
```

Comment:
simply use ```if (reco::deltaR2(cand, muon) > dR2_threshold ) continue;``` instead of the code in L873-879.

Also, please drop extra spaces in front of the terminating semicolons

Reply:
@slava77  done! #26233